<?xml version="1.0"?>
<launch>
    <arg name="argID"           default="0"/>
    <arg name="argVehicle"      default="standard_vtol"/>
    <arg name="argNamespace"    default="$(arg argVehicle)$(arg argID)"/>

    <arg name="argNodeRate"         default="100.0"/>
    <arg name="argPosError"         default="2.0"/>
    <arg name="argYawError"         default="10.0"/>

    <arg name="argSmoothFactor"         default="3.0"/>
    <arg name="argMaxVelocity"          default="4.0"/>
    <arg name="argReachDistance"        default="8.0"/>
    <arg name="argMaxAngularVelocity"   default="45.0"/>
    <arg name="argForceLinearSpeed"     default="0.04"/>

    <node name="navigation_controller_node" type="navigation_controller_node" pkg="navigation_controller" output="screen" required="true" ns="$(arg argNamespace)">
        <param name="paramNodeRate"             type="double" value="$(arg argNodeRate)"/>
        <param name="paramPosError"             type="double" value="$(arg argPosError)"/>
        <param name="paramYawError"             type="double" value="$(arg argYawError)"/>
        <param name="paramMaxVelocity"          type="double" value="$(arg argMaxVelocity)"/>
        <param name="paramSmoothFactor"         type="double" value="$(arg argSmoothFactor)"/>
        <param name="paramReachDistance"        type="double" value="$(arg argReachDistance)"/>
        <param name="paramForceLinearSpeed"     type="double" value="$(arg argForceLinearSpeed)"/>
        <param name="paramMaxAngularVelocity"   type="double" value="$(arg argMaxAngularVelocity)"/>
    </node>
</launch>
